idaes/apps/grid_integration/bidder.py

```diff
@@ -1291,3 +1291,395 @@ def _record_bids(self, bids, date, hour, **kwargs):
+        self._check_inputs()
+
+        self.generator = self.bidding_model_object.model_data.gen_name
+        self.bids_result_list = []
+
+    @property
+    def generator(self):
+        return self._generator
+
+    @generator.setter
+    def generator(self, name):
+        self._generator = name
+
+    def formulate_DA_bidding_problem(self):
+        """
+        No need to formulate a DA bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def formulate_RT_bidding_problem(self):
+        """
+        No need to formulate a RT bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def compute_day_ahead_bids(self, date, hour=0):
+        raise NotImplementedError
+
+    def compute_real_time_bids(
+        self,
+        date,
+        hour,
+        realized_day_ahead_prices,
+        realized_day_ahead_dispatches,
+        tracker_profile,
+    ):
+        raise NotImplementedError
+
+    def update_day_ahead_model(self, **kwargs):
+        """
+        No need to update the RT bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def update_real_time_model(self, **kwargs):
+        """
+        No need to update the RT bidding problem here.
+
+        Arguments:
+            None
+
+        Returns:
+            None
+        """
+        pass
+
+    def record_bids(self, bids, model, date, hour, market):
+        """
+        This function records the bids and the details in the underlying bidding model.
+
+        Arguments:
+            bids: the obtained bids for this date.
+
+            model: bidding model
+
+            date: the date we bid into
+
+            hour: the hour we bid into
+
+        Returns:
+            None
+
+        """
+
+        # record bids
+        self._record_bids(bids, date, hour, Market=market)
+
+        # record the details of bidding model
+        for i in model.SCENARIOS:
+            self.bidding_model_object.record_results(
+                model.fs[i], date=date, hour=hour, Scenario=i, Market=market
+            )
+
+        return
+
+    def _record_bids(self, bids, date, hour, **kwargs):
+        """
+        Record the bis of each time perid.
+
+        Arguments:
+            bids: the obtained bids for this date.
+
+            date: the date we bid into
+
+            hour: the hour we bid into
+
+        Returns:
+            None
+
+        """
+        df_list = []
+        for t in bids:
+            for gen in bids[t]:
+
+                result_dict = {}
+                result_dict["Generator"] = gen
+                result_dict["Date"] = date
+                result_dict["Hour"] = t
+
+                for k, v in kwargs.items():
+                    result_dict[k] = v
+
+                pair_cnt = len(bids[t][gen]["p_cost"])
```

Comment:
What does `pair_cnt` stand for? The number of bid elements? Perhaps change to `num_bid_pairs` as the variable name?

Reply:
Done